zoo/linkedlist/CustomLinkedList.java

```diff
@@ -0,0 +1,95 @@
+package zoo.linkedlist;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class CustomLinkedList<T> {
+
+    private static final Logger LOGGER = LogManager.getLogger(CustomLinkedList.class);
+    private CustomNode<T> head;
+
+    public CustomLinkedList() {
+
+        this.head = null;
+
+    }
+
+    public void addNode(CustomNode<T> node) {
+
+        if (this.head == null) {
+
+            this.head = node;
+        }
+
+        else {
+            CustomNode<T> currentNode = this.head;
+            while (currentNode.next != null) {
+
+                currentNode = currentNode.next;
+            }
+            currentNode.next = node;
+
+        }
+
+        LOGGER.info("Node added");
+    }
+
+    public void removeNode(CustomNode<T> node) {
+
+        CustomNode<T> previousNode = null;
+        CustomNode<T> currentNode = this.head;
+        CustomNode<T> nextNode = null;
+
+        if (this.head == node) {
+
+            if (node.next == null) {
+
+                this.head = null;
+            }
+
+            else {
+
+                this.head.next = node.next;
+            }
+        }
```

Comment:
please remove unnecessary empty lines is the code, here is an example: 
```suggestion
    public void removeNode(CustomNode<T> node) {
        CustomNode<T> previousNode = null;
        CustomNode<T> currentNode = this.head;
        CustomNode<T> nextNode = null;
        if (this.head == node) {
            if (node.next == null) {
                this.head = null;
            } else {
                this.head.next = node.next;
            }
        }
```